### PR TITLE
[wasm] Fix provision-wasm to do emsdk selection. Remove the need for the emcc patch.

### PR DESCRIPTION
**Pull request by kumpera**

### CODE REVIEW

File: sdks/builds/wasm.mk [hunk 1]
```diff
@@ -6,12 +6,19 @@ EMSCRIPTEN_SDK_DIR=$(TOP)/sdks/builds/toolchains/emsdk
...
+
+.stamp-wasm-checkout-and-update-emsdk: | $(EMSCRIPTEN_SDK_DIR)
+	cd $(TOP)/sdks/builds/toolchains/emsdk && git pull
+	touch $@
+
+.stamp-wasm-install-and-select-the-right-emsdk: .stamp-wasm-checkout-and-update-emsdk
 	cd $(TOP)/sdks/builds/toolchains/emsdk && ./emsdk install sdk-$(EMSCRIPTEN_VERSION)-64bit
 	cd $(TOP)/sdks/builds/toolchains/emsdk && ./emsdk activate --embedded sdk-$(EMSCRIPTEN_VERSION)-64bit
-	-cd $(TOP)/sdks/builds/toolchains/emsdk/emscripten/$(EMSCRIPTEN_VERSION) && patch -p1 < $(TOP)/sdks/builds/emsdk-eh.diff
+	-cd $(TOP)/sdks/builds/toolchains/emsdk/emscripten/$(EMSCRIPTEN_VERSION) && patch -N -p1 < $(TOP)/sdks/builds/emsdk-eh.diff
```
Comment (akoeplinger) [inline sdks/builds/wasm.mk]: this patch was deleted

File: sdks/builds/wasm.mk [hunk 2]
```diff
@@ -6,12 +6,19 @@ EMSCRIPTEN_SDK_DIR=$(TOP)/sdks/builds/toolchains/emsdk
 
 $(TOP)/sdks/builds/toolchains/emsdk:
 	git clone https://github.com/juj/emsdk.git $(EMSCRIPTEN_SDK_DIR)
+
+.stamp-wasm-checkout-and-update-emsdk: | $(EMSCRIPTEN_SDK_DIR)
```
Comment (akoeplinger) [inline sdks/builds/wasm.mk]: don't you need to somehow include the version in the stamp file? otherwise on a bump you'd not get the new version

**Comment by luhenry** [comment]
@monojenkins squash

**Comment by lewurm** [comment]
@monojenkins build Linux x64

**Pull request merged**